src/offspot_demo/utils/systemd.py

```diff
@@ -0,0 +1,85 @@
+import subprocess
+
+from offspot_demo.utils.process import run_command
+
+
+class SystemdError(Exception):
+
+    def __init__(self, stdout: str, *args: object) -> None:
+        self.stdout = stdout
+        super().__init__(*args)
+
+
+class SystemdNotRunningError(SystemdError):
+    pass
+
+
+class SystemdNotWaitingError(SystemdError):
+    pass
+
+
+class SystemdNotEnabledError(SystemdError):
+    pass
+
+
+class SystemdNotLoadedError(SystemdError):
+    pass
+
+
+def check_systemd_service(
+    unit_fullname: str,
+    *,
+    check_running: bool = False,
+    check_waiting: bool = False,
+    check_enabled: bool = False,
+) -> subprocess.CompletedProcess[str]:
+    """Check status of the systemd unit
+
+    By default, check at least that the unit is loaded properly (i.e. parsing is ok),
```

Comment:
The convention regarding this (so it an be picked up by tools such as mkdocs) is:

```
Parameters:
    param_name: description

Raises:
    exception: details
```

```
Raises:
    SystemdNotLoadedError: if unit is not properly loaded
    SystemdNotRunningError: if unit is not running ; when using check_running
    SystemdNotEnabledError: if unit is not enabled ; when using check_enabled
    SystemdNotWaitingError: if unit is not waiting ; when using check_waiting
```